core/modules/layout/plugins/context/entity_layout_context_handler.inc

```diff
@@ -24,15 +24,12 @@ class EntityLayoutContext extends LayoutContext {
 
   function label() {
     $info = entity_get_info($this->entity_type);
-    // @todo: Contexts do not yet have settings to make this possible. Once
-    // the ability to add manual contexts (e.g. an arbitrary node), then this
-    // will actually do something.
-    if (isset($this->settings['id'])) {
-      if ($entity = entity_load($this->entity_type, $this->settings['id'])) {
+    if (isset($this->settings['entity'])) {
+      if ($entity = entity_load($this->entity_type, $this->settings['entity'])) {
         return t('@type: @id (@title)', array('@type' => $info['label'], '@id' => $entity->id(), '@title' => $entity->label()));
       }
       else {
-        return t('Unknown @type: @id', array('@type' => $info['label'], '@id' =>$this->settings['id']));
+        return t('Unknown @type: @id', array('@type' => $info['label'], '@id' =>$this->settings['entity']));
```

Comment:
Maybe @id needs to change name too. Though not clear here what kind of variable "entity" is.

Reply:
This needs more thinking actually. 

For now we're loading data into contexts with the following code:
```
          $function = $context_info['load callback'];
          $context_data = $function($context->settings['entity']); <-- this is really the entity ID
          $context->setData($context_data);
```
It would be better to pass all of `settings` to the `load callback` I think, to accommodate any contrib context types with more complex settings than just an `ID`, but the existing context types just use the object loaders as the `load callback`, so we'd need to change their callbacks to a generic function that accepts just `settings` then in that function call `ENTITY_load($settings['entity'])`.

I think I need @quicksketch opinion.

Reply:
Resolved in latest commits.